tasks/managed/filter-already-released-images/filter-already-released-images.yaml

```
@@ -0,0 +1,344 @@
+
+        FILTERED_COMPONENTS='[]'
+        FILTERED_COUNT=0
+
+        for ((i=0; i<COMPONENT_COUNT; i++)); do
+            COMPONENT=$(jq -c ".components[$i]" <<< "${SNAPSHOT_JSON}")
+            COMPONENT_NAME=$(jq -r '.name' <<< "${COMPONENT}")
+            CONTAINER_IMAGE=$(jq -r '.containerImage' <<< "${COMPONENT}")
+
+            # Get the source image digest using oras resolve (same as push-snapshot)
+            # This ensures we compare manifest index digests, not platform-specific ones
+            SOURCE_AUTH_FILE=$(mktemp)
+            if ! select-oci-auth "${CONTAINER_IMAGE}" > "${SOURCE_AUTH_FILE}" 2>/dev/null || \
+               [ ! -s "${SOURCE_AUTH_FILE}" ]; then
+                echo '{}' > "${SOURCE_AUTH_FILE}"
+            fi
+
+            SOURCE_ERROR_FILE=$(mktemp)
+            if ! DIGEST=$(oras resolve --registry-config "${SOURCE_AUTH_FILE}" \
+                "${CONTAINER_IMAGE}" 2>"${SOURCE_ERROR_FILE}"); then
+                SOURCE_ERROR_MSG=$(cat "${SOURCE_ERROR_FILE}")
+                rm -f "${SOURCE_ERROR_FILE}" "${SOURCE_AUTH_FILE}"
+                
+                # Check if it's a benign error (image not found in registry)
+                if is_not_found_error "${SOURCE_ERROR_MSG}"; then
+                    # Treat as "not yet released", keep component for release
+                    echo "WARNING: Source image ${CONTAINER_IMAGE} not found in registry, treating as not yet released"
+                    FILTERED_COMPONENTS=$(jq --argjson comp "${COMPONENT}" '. += [$comp]' <<< "${FILTERED_COMPONENTS}")
+                    continue
+                fi
+                
+                # Critical error (auth, network, etc) - fail the task
+                echo "ERROR: Failed to resolve source image ${CONTAINER_IMAGE}: ${SOURCE_ERROR_MSG}"
+                exit 1
+            fi
+            rm -f "${SOURCE_ERROR_FILE}" "${SOURCE_AUTH_FILE}"
+
+            if [ -z "${DIGEST}" ]; then
```

Won't this be true if we hit the condition on line 171? Thus exiting with error anyways

---

No, we won't reach line 175 if we hit the condition on line 162. (line numbers are different from the push commented initially; they were changed after some code modification, sorry; I'm using the updated values)

Line 175's purpose is to catch the weird edge case where oras resolve returns success (exit code 0) but somehow outputs an empty string instead of a digest. 
This would be a bug in oras itself, but we want to catch and fail explicitly rather than silently continuing with an empty digest.
So the two checks protect against different scenarios:
Line 162: Handle expected "not found" errors gracefully (keep component)
Line 175: Catch unexpected behavior (successful command with no output) as a hard error

---

thank you for explaining